### PR TITLE
add js client lib

### PR DESCRIPTION
**Pull request by asurkov**
feedback-ready, not feature-complete for sure :)

### CODE REVIEW

File: js/src/LogEvent.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
+(function(exports) {
```
Comment (sergey48k) [inline js/src/LogEvent.js:4]: Copyright

File: js/src/LogEvent.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+ */
+class LogEvent
+{
+  constructor(url, { headers })
+  {
+    console.assert(url);
+    console.assert(headers);
+
+    this.url = url;
+    if (headers['set-cookie']) { // got set-cookie
```
Comment (sergey48k) [inline js/src/LogEvent.js]: Our standard comment style is: ```[two spaces]// Full sentence with dot.```

File: js/src/LogEvent.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+   */
+  event(type, fields)
+  {
+    let data = Object.assign({
+      session: this.session,
+      type,
+      timestamp: Date.now()
+    }, fields);
+
+    let thisobj = this;
```
Comment (sergey48k) [inline js/src/LogEvent.js:44]: ```let self = this;``` is a bit shorter.

File: js/src/LogEvent.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    });
+    req.end();
+  }
+
+  browserRequest(data, resolve, reject)
+  {
+    const req = new XMLHttpRequest();
+    req.open('POST', this.url, true);
+
+    //Send the proper header information along with the request
```
Comment (sergey48k) [inline js/src/LogEvent.js]: ```//[space]Send ... request.```

**Review by sergey48k — APPROVED** [review]
lgtm but this needs tests

File: js/src/LogEvent.js (hunk 3 above)
Comment (asurkov) [reply js/src/LogEvent.js:44]: I would avoid using self because this is a global property (https://developer.mozilla.org/en-US/docs/Web/API/Window/self)

**Pull request merged**